libfwupd/fwupd-client-sync.c

```diff
@@ -913,16 +913,16 @@ fwupd_client_modify_bios_attr_cb(GObject *source, GAsyncResult *res, gpointer us
  **/
 gboolean
 fwupd_client_modify_bios_attr(FwupdClient *self,
-			      const gchar *key,
-			      const gchar *value,
+			      GPtrArray *keys,
+			      GPtrArray *values,
```

Comment:
hmm, this doesn't feel very idiomatic. Could we do a GHashTable of key/values? Was there a huge overhead with `n`xDBus requests from the client? The single key/value sure seems a lot easier to use.

Reply:
The problem with multiple requests from the client is that you need to authenticate each one of them.  So you need some sort of way to send all the pairs atomically to avoid that.

I think a hashtable still will have the same problem - at the end of the day you need to convert to string to pass over dbus don't you?